core/util.go

```diff
@@ -95,6 +96,19 @@ func Fingerprint256(data []byte) string {
 	return B64enc(d.Sum(nil))
 }
 
+func KeyDigest(key crypto.PublicKey) string {
+	switch t := key.(type) {
+		case *jose.JsonWebKey:
+			return KeyDigest(t.Key)
+		case jose.JsonWebKey:
+			return KeyDigest(t.Key)
+		default:
+			keyDER, _ := x509.MarshalPKIXPublicKey(key)
```

Comment:
Good point. I can add an error return value, and provide a convenience KeyDigestEquals method that returns false if either key has a marshaling error.